File: ServerTest/Context/Test/src/main/java/org/elkoserver/server/context/test/TestInternalObject.java
package org.elkoserver.server.context.test;

import org.elkoserver.foundation.json.JSONMethod;
import org.elkoserver.foundation.json.MessageHandlerException;
import org.elkoserver.json.JSONLiteral;
import org.elkoserver.server.context.AdminObject;
import org.elkoserver.server.context.InternalActor;

import static org.elkoserver.json.JSONLiteralFactory.targetVerb;

class TestInternalObject extends AdminObject {
    /**
     * JSON-driven constructor.
     */
    @JSONMethod
    public TestInternalObject() {
        super();
    }

    @JSONMethod({ "arg" })
    public void boom(InternalActor from, String arg)
            throws MessageHandlerException
    {
        JSONLiteral response = targetVerb(this, "bah");
        response.addParameter("arg", arg);
        response.finish();
        from.send(response);
    }

    @JSONMethod({ "arg" })
    public void superboom(InternalActor from, String arg)
            throws MessageHandlerException
    {
        from.ensureAuthorized();
        JSONLiteral response = targetVerb(this, "superbah");
        response.addParameter("arg", arg);
        response.finish();
        from.send(response);
    }
}

File: ServerTest/Context/Test/src/main/kotlin/org/elkoserver/server/context/test/TestInternalObject.kt
package org.elkoserver.server.context.test

import org.elkoserver.foundation.json.JsonMethod
import org.elkoserver.json.JsonLiteralFactory
import org.elkoserver.server.context.AdminObject
import org.elkoserver.server.context.InternalActor

internal class TestInternalObject @JsonMethod constructor() : AdminObject() {
    @JsonMethod("arg")
    fun boom(from: InternalActor, arg: String) {
        val response = JsonLiteralFactory.targetVerb(this, "bah").apply {
            addParameter("arg", arg)
            finish()
        }
        from.send(response)
    }

    @JsonMethod("arg")
    fun superboom(from: InternalActor, arg: String) {
        from.ensureAuthorized()
        val response = JsonLiteralFactory.targetVerb(this, "superbah").apply {
            addParameter("arg", arg)
            finish()
        }
        from.send(response)
    }
}
